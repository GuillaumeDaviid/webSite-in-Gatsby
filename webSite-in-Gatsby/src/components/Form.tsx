import * as React from 'react'
import './styles/Form.scss'
import { useState } from 'react'
import { useGoogleReCaptcha } from 'react-google-recaptcha-v3'

export default function Form() {
    return(
        <div className="Form">
            <section id="contact">
            <div className="Form_text">
             
            <h2>DEMANDEZ UN DEVIS</h2>
            <div className="Form_text-separator"></div>
            <p className="Form_text-content">Envie d&apos;un nouveau site web ? Ou alors d&apos;une refonte de celui-ci ? 
            Vous cherchez un développeur qui répond à vos besoins sur-mesure et qui respect les délais et le budget ? Demandez un Devis !</p>
            
            </div>

            <div className="Form_content">
                
            <div className="Form_content-info">
           
                <h4>COORDONNEES</h4>
                <p>Loos-en-Gohelle (62) - France</p>
                <p>guillaumedaavidpro@gmail.com</p>
                <p>Guillaume David</p>
                <a>Twitter</a>
                <a>github</a>
                <a>medium</a>
                <a>linkedin</a>
               
            </div>
            

            <div className="Form_content-form">
                <form id="formid" method="post" action='https://getform.io/f/ca7fb4f1-379c-4242-a03a-93dd5d1d0078' /*ou https://www.guillaumedavid.com/api/recaptcha.php  ../api/recaptcha.php */>
                    <div className="row_form">

                    <div >
                        <input className="row_form-name" type="text" name="user_name" placeholder="Nom"/>
                    </div>

                    <div >
                        <input className="row_form-mail" type="email" name="user_mail" placeholder="Email"/>
                    </div>

                    </div>

                    <div>
                        <textarea className="row_form-msg" name="user_message" placeholder="Message"/>
                    </div>

                    <div className="container-btn">
                    <input type="submit" value="ENVOYER" className="btn-form g-recaptcha"/>
                    
                    </div>
                </form>
            </div>

            </div>
            </section>
        </div>

        
    )
}