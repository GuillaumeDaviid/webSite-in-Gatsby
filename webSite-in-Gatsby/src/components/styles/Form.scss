.Form {
    &_text {
        text-align: center;
        display: flex;
        justify-content: center;
        align-items: center;
        flex-direction: column;

        &-separator {
            height: 2px;
            width: 85px;
            margin: 70px auto;
            background-color: #222222;
        }

        &-content {
            font-size: 1.375rem;
            color: #222222;
            line-height: 2rem;
            width: 75%;
            margin: 0 auto;
            font-family: 'amblelight';


        }
    }

    &_content {
        margin-top: 100px;
        display: flex;

        &-info {
            overflow-wrap: break-word;
            background: #f2f2f2;
            padding: 90px 50px;
            width: 34%;
            font-style: italic;
            color: #222222;
            display: inline-block;
            font-size: 1.375rem;
            font-family: 'amblelight';
        }

        &-form {
            padding: 120px 80px;
            background-color: #e4e4e4;
            width: 66%;
        }

        &-network {
            display: flex;
            justify-content: space-between;
            width: 25vw;
        }
    }
}

.row_form {
    display: flex;
    gap:20vw;
    
    &-name, &-mail, &-msg {
        display: inline-block;
        width: 100%;
        height: 35px;
        padding: 6px 25px;
        font-size: 14px;
        line-height: 1.42857143;
        // color: #d87843;
        color: black;
        background-color: transparent;
        background-image: none;
        border: 0px solid #ccc;
        border-bottom: 2px solid #ddd;

        &:focus {
            color: #d87843;
            border-color: #d87843;
            -webkit-box-shadow: inset 0 0px 0px rgb(0 0 0 / 8%), 0 0 0px rgb(102 175 233 / 60%);
            box-shadow: inset 0 0px 0px rgb(0 0 0 / 8%), 0 0 0px rgb(102 175 233 / 60%);
            outline: none;   
        }
    }

    &-msg {
        margin-top: 50px;
        height: 70px;
    }
}

.btn-form {
    text-align: center;
    letter-spacing: 1px;
    font-size: 1.875rem;
    border-radius: 0px;
    margin-top: 5px;
    background-color: transparent;
    transition: .5s;
    font-family: 'amblebold';
    border: 2px solid #000;
    color: #222;
    padding: 0.4rem 2rem;
    height: 60px;
    cursor: pointer;
    font-weight: 999;

    &:hover {
        color: white;
        background-color: #d87843;
    }
}

.container-btn {
    display: flex;
    justify-content: space-between;
    margin-top: 10px;
}

@media (max-width: 750px) {
    .Form {
        &_content {
            display: flex;
            flex-direction: column;

            &-form {
                width: 100%;
            }

            &-info {
                width: 100%;
            }
        }
    }
}

@media (max-width: 500px) {
    .Form {
        &_text {
            &-content {
                font-size: 1rem;
            }
        }

        &_content {
            &-info {
                font-size: 1rem;
            }
        }
    }

    .row_form {
        display: flex;
        flex-direction: column;
    }

    .btn-form {
        font-size: 1rem;
    }
}