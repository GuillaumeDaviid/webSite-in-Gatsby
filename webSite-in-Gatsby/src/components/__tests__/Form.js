import React from "react"
import renderer from "react-test-renderer"
import Form from "../Form.tsx"
import { render, screen } from "@testing-library/react"

describe("Form", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Form />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a title element", () => {
    render(<Form />)

    const title = screen.getByTestId('title');
    expect(title).toBeTruthy();
  })

  it("should have a separator element", () => {
    render(<Form />)

    const separator = screen.getByTestId('separator');
    expect(separator).toBeTruthy();
  })

  it("should have a text element", () => {
    render(<Form />)

    const text = screen.getByTestId('text');
    expect(text).toBeTruthy();
  })

  it("should have a form element", () => {
    render(<Form />)

    const form = screen.getByTestId('form');
    expect(form).toBeTruthy();
  })

  it("should have a info element", () => {
    render(<Form />)

    const info = screen.getByTestId('info');
    expect(info).toBeTruthy();
  })

  it("should have a reCaptcha element", () => {
    render(<Form />)

    const reCaptcha = screen.getByTestId('reCaptcha');
    expect(reCaptcha).toBeTruthy();
  })
})