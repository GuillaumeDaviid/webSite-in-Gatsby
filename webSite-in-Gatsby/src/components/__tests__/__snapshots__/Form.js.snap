// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<div
  className="Form"
>
  <section
    id="contact"
  >
    <div
      className="Form_text"
    >
      <h2
        data-testid="title"
      >
        DEMANDEZ UN DEVIS
      </h2>
      <div
        className="Form_text-separator"
        data-testid="separator"
      />
      <p
        className="Form_text-content"
        data-testid="text"
      >
        Envie d'un nouveau site web ? Ou alors d'une refonte de celui-ci ? Vous cherchez un développeur qui répond à vos besoins sur-mesure et qui respect les délais et le budget ? Demandez un Devis !
      </p>
    </div>
    <div
      className="Form_content"
    >
      <div
        className="Form_content-info"
        data-testid="info"
      >
        <h3>
          COORDONNEES
        </h3>
        <p>
          Loos-en-Gohelle (62) - France
        </p>
        <p>
          guillaumedaavidpro@gmail.com
        </p>
        <p>
          Guillaume David
        </p>
        <div
          className="Form_content-network"
        >
          <a
            href="https://github.com/GuillaumeDaviid"
            target="_blank"
          >
            github
          </a>
          <a
            href="https://guillaumedaavid.medium.com/"
            target="_blank"
          >
            medium
          </a>
          <a
            href="https://www.linkedin.com/in/guillaume-david-5541271b9/"
            target="_blank"
          >
            linkedin
          </a>
        </div>
      </div>
      <div
        className="Form_content-form"
        data-testid="form"
      >
        <form
          action="https://getform.io/f/ca7fb4f1-379c-4242-a03a-93dd5d1d0078"
          id="formid"
          method="post"
        >
          <div
            className="row_form"
          >
            <div>
              <input
                className="row_form-name"
                id="name"
                name="name"
                placeholder="Nom"
                required={true}
                type="text"
              />
            </div>
            <div>
              <input
                className="row_form-mail"
                name="email"
                placeholder="Email"
                required={true}
                type="email"
              />
            </div>
          </div>
          <div>
            <textarea
              className="row_form-msg"
              name="message"
              placeholder="Message"
              required={true}
            />
          </div>
          <div
            className="container-btn"
          >
            <div
              data-testid="reCaptcha"
            />
            <input
              className="btn-form"
              type="submit"
              value="ENVOYER"
            />
          </div>
        </form>
      </div>
    </div>
  </section>
</div>
`;
